Firestore/core/src/model/values.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_MODEL_VALUES_H_
+#define FIRESTORE_CORE_SRC_MODEL_VALUES_H_
+
+#include <string>
+
+#include "Firestore/Protos/nanopb/google/firestore/v1/document.nanopb.h"
+
+namespace firebase {
+namespace firestore {
+
+namespace util {
+enum class ComparisonResult;
+}
+
+namespace model {
+
+/**
+ * The order of types in Firestore. This order is based on the backend's
+ * ordering, but modified to support server timestamps.
+ */
+const int32_t kTypeOrderNull = 0;
+const int32_t kTypeOrderBoolean = 1;
+const int32_t kTypeOrderNumber = 2;
+const int32_t kTypeOrderTimestamp = 3;
+const int32_t kTypeOrderServerTimestamp = 4;
+const int32_t kTypeOrderString = 5;
+const int32_t kTypeOrderBlob = 6;
+const int32_t kTypeOrderReference = 7;
+const int32_t kTypeOrderGeoPoint = 8;
+const int32_t kTypeOrderArray = 9;
+const int32_t kTypeOrderMap = 10;
+
+class Values {
```

Comment:
This file contains equals, compareTo and canonical ID support. That's all we have on Web and Android, but in general we treated this file as a "proto values utility" class on the other platforms.

Reply:
Nit: a file named `value` seems to indicate that it defines a `Value` class. I don't have a better suggestion than the uninspired `value_helpers`, but personally, I feel it would convey the intention better (same for the server timestamp one). If you can think of a better name, that's great.